dbt/include/athena/macros/adapters/relation.sql

```diff
@@ -1,16 +1,10 @@
 {% macro athena__drop_relation(relation) -%}
-  {% set rel_type = adapter.get_table_type(relation.schema, relation.table) %}
+  {% set rel_type = adapter.get_table_type(relation.schema, relation.table).value %}
   {%- if rel_type is not none %}
     {%- if rel_type == 'table' %}
       {%- do adapter.clean_up_table(relation.schema, relation.table) -%}
     {%- endif %}
-    {% call statement('drop_relation', auto_begin=False) -%}
-      {%- if relation.type == 'view' -%}
-        drop {{ relation.type }} if exists {{ relation.render() }}
-      {%- else -%}
-        drop {{ relation.type }} if exists {{ relation.render_hive() }}
-      {% endif %}
-    {%- endcall %}
+    {%- do adapter.delete_from_glue_catalog(relation) -%}
```

Comment:
@chrischin478 what happens with iceberg table's data in case when it's droped via glue catalog ? it should be dropped as well I believe. I've tried to create iceberg tables via glue API but there are problems with metadata files. Are there similar problems when you drop table ?

Reply:
Good point indeed, I think that there is something missing in the pr. 

Iceberg tables are managed, therefore when you drop a table it also delete the data from s3. here the issue, when you have quite some objects, the drop statement leave the s3 path not clean.
To make the current implementation working we need also to add another check for iceberg tables
```
{%- if rel_type == 'table'  or rel_type== 'iceberg_table' %}
{%- do adapter.clean_up_table(relation.schema, relation.table) -%}
```

Reply:
It seems to overkill for iceberg table. I would prefer to use native logic with running `drop table if exists ...` query for iceberg tables

Reply:
I agree with @nicor88 , the native Iceberg delete when there is too much data or metadata will lead to inconsistency and therefore the data can't be deleted afterwards (without going by the delete in S3 and in glue). Therefore we can:
- do the Athena delete for views
- do the S3 prune + glue delete for tables (no matter what strategy / table type)
Am I missing something ?

Reply:
Isn't that covered by this change?

In the case of views we just delete via API call, for tables (iceberg included, currently missing) we prune and delete the table via glue api.